Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -134,6 +136,76 @@ TEST_F(SerializerTest, EncodesBoolProtoToBytes) {
   }
 }
 
+TEST_F(SerializerTest, EncodesIntegersModelToProto) {
+  std::list<int64_t> testCases{0,
+                               1,
+                               -1,
+                               100,
+                               -100,
+                               std::numeric_limits<int64_t>::min(),
+                               std::numeric_limits<int64_t>::max()};
+  for (int64_t test : testCases) {
+    FieldValue model = FieldValue::IntegerValue(test);
+    Serializer::TypedValue proto{FieldValue::Type::Long,
+                                 google_firestore_v1beta1_Value_init_default};
+    proto.value.integer_value = test;
+    ExpectRoundTrip(model, proto, FieldValue::Type::Long);
+  }
+}
+
+TEST_F(SerializerTest, EncodesIntegersProtoToBytes) {
+  // NB: because we're calculating the bytes via protoc (instead of
+  // libprotobuf) we have to look at values from numeric_limits<T> ahead of
+  // time. So these test cases make the following assumptions about
+  // numeric_limits: (linking libprotobuf is starting to sound like a better
```

Comment:
FWIW I agree but don't want to necessarily derail you.

Reply:
Yup. I think I'll experiment with this shortly. (Though I may wait for a few more PRs that pile on the tech dept until it becomes impossible to ignore any further.)